---
title: "Sujan_Drought"
output: html_document
date: "2024-06-25"
---

```{r}
source("https://raw.githubusercontent.com/joey711/phyloseq/master/inst/scripts/installer.R",
       local = TRUE)

install_phyloseq(branch = "devel")

library("phyloseq")
library("ggplot2")
library("scales")
library("ape")
library("vegan")
library("data.table")
library("RColorBrewer")
library("colorRamps")
library("VennDiagram")
library("beepr")
library("agricolae")
library("seqinr")
library("patchwork")
library("labdsv")
library("reshape2")
library("dplyr")

install.packages("pacman")

pacman::p_load(readxl, tidyverse, conflicted, lme4, lmerTest, emmeans, multcomp, multcompView, ggplot2, desplot, ggsci, ggrepel)\

install.packages("pairwiseAdonis")
install.packages("rstatix")
install.packages("randomForest")
install.packages("rfUtilities")
install.packages("car")
install.packages("caret")
install.packages("ROCR")
install.packages("pdp")
install.packages("RColorBrewer")



library(phyloseq)
library(stats)
library(FSA)
library(vegan)
library(pairwiseAdonis)
library(ggplot2)
library(rstatix)
library(randomForest)
library(rfUtilities)
library(dplyr)
library(car)
library(caret)
library(Maaslin2, lib.loc="/home/ubuntu/CourseData/MIC_data/.conda/envs/r-env/lib/R/library")
library(ROCR)
library(pdp)
library(RColorBrewer)
```

##Importing data

```{r}
biom = import_biom("sujan_16s.biom")

metadata = import_qiime_sample_data("metadata.txt")

tree = read_tree("rooted-tree-sujan16s.nwk")

sujan_biom = merge_phyloseq(biom, metadata, tree)



#rename columns in taxonomy table
colnames(tax_table(sujan_biom)) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")
colnames(tax_table(sujan_biom))
```


```{r}
#a function to get indicator species for levels of a given column in your metadata file
#takes a phyloseq object and returns a dataframe of indicators
get_indicators <- function(physeq, groupfactor){
all.data <- psmelt(physeq)
all.data <- subset(all.data, Abundance > 0)
metadata <- sample_data(physeq)

spec <- subset(all.data, select = c("Sample", "OTU", "Abundance"))
spec <- dcast(spec, Sample ~ OTU, value.var =  "Abundance")
spec <- mutate_at(spec, c(2:ncol(spec)), ~replace(., is.na(.), 0))
rownames(spec) <- spec$Sample
spec <- subset(spec, select = -c(Sample))
spec <- spec[order(rownames(spec)),]

clust <- as.data.frame(metadata[[groupfactor]])
colnames(clust) <- "group"
clust$group.level <- as.integer(clust$group)
rownames(clust) <- metadata$SampleID
clust <- subset(clust, rownames(clust) %in% rownames(spec))
clust <- clust[order(rownames(clust)),]

if (!(identical(rownames(spec), rownames(clust)))) {
  stop("Species and clustering dataframes are misaligned! Check sample orders.")
}

indvals <- indval(spec, clust$group.level, numitr = 1000)
summary(indvals)

#convert all.indvals to a single df for ease of access
pval <- as.data.frame(indvals$pval)
pval$ASV <- rownames(pval)
maxcls <- as.data.frame(indvals$maxcls)
maxcls$ASV <- rownames(maxcls)
indcls <- as.data.frame(indvals$indcls)
indcls$ASV <- rownames(indcls)
all.indvals <- merge(pval, maxcls, by = "ASV")
all.indvals <- merge(all.indvals, indcls, by = "ASV")
colnames(all.indvals) <- c("ASV", "pval", "group.level", "indcls")
all.indvals <- merge(all.indvals, clust, by = "group.level")
all.indvals <- subset(all.indvals, select = c(ASV, group, pval, indcls))
all.indvals <- unique(all.indvals)

#get only significant indicators (p<=0.05) with an indcls >=0.5
sig.indvals <- subset(all.indvals, pval <= 0.05)
sig.indvals <- subset(sig.indvals, indcls >= 0.5)

#return an object with both the subsetted significant indvals and all indvals
indvals <- list("all.indvals" = all.indvals, "sig.indvals" = sig.indvals, "all.data" = indvals)
}

#a function to make a fasta file of just indicator species
#requires original repseqs fasta file and a list of indicators generated by get_indicators
#NOTE--this writes the fasta file out to your working directory
make_fasta <- function(indicator.list, repseqs, outfile){
  #subset repseqs file to indicators only
  indicators <- repseqs[names(repseqs) %in% unique(indicator.list$ASV)]
  #write it out to file to run through fasttree
  indicator.names <- names(indicators)
  indicator.sequences <- getSequence(indicators)
  write.fasta(indicator.sequences, indicator.names, outfile)
}
```


```{r}
#make a clean taxonomy table
clean_taxtable <- function(physeq){
  tax.table <- as.data.frame(tax_table(physeq))
  colnames(tax.table) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
  tax.table$Kingdom <- as.character(lapply(tax.table$Kingdom, sub, pattern = "k__", replacement = ""))
  tax.table$Phylum <- as.character(lapply(tax.table$Phylum, sub, pattern = "p__", replacement = ""))
  tax.table$Class <- as.character(lapply(tax.table$Class, sub, pattern = "c__", replacement = ""))
  tax.table$Order <- as.character(lapply(tax.table$Order, sub, pattern = "o__", replacement = ""))
  tax.table$Family <- as.character(lapply(tax.table$Family, sub, pattern = "f__", replacement = ""))
  tax.table$Genus <- as.character(lapply(tax.table$Genus, sub, pattern = "g__", replacement = ""))
  tax.table$Species <- as.character(lapply(tax.table$Species, sub, pattern = "s__", replacement = ""))
  tax.table$Kingdom <- ifelse(tax.table$Kingdom == "", "Unknown", as.character(tax.table$Kingdom))
  tax.table$Phylum <- ifelse(tax.table$Phylum == "", "Unknown", as.character(tax.table$Phylum))
  tax.table$Class <- ifelse(tax.table$Class == "", "Unknown", as.character(tax.table$Class))
  tax.table$Order <- ifelse(tax.table$Order == "", "Unknown", as.character(tax.table$Order))
  tax.table$Family <- ifelse(tax.table$Family == "", "Unknown", as.character(tax.table$Family))
  tax.table$Genus <- ifelse(tax.table$Genus == "", "Unknown", as.character(tax.table$Genus))
  tax.table$Species <- ifelse(tax.table$Species == "", "Unknown", as.character(tax.table$Species))
  tax.table$Kingdom <- ifelse(is.na(tax.table$Kingdom) == TRUE , "Unknown", as.character(tax.table$Kingdom))
  tax.table$Phylum <- ifelse(is.na(tax.table$Phylum) == TRUE , "Unknown", as.character(tax.table$Phylum))
  tax.table$Class <- ifelse(is.na(tax.table$Class) == TRUE , "Unknown", as.character(tax.table$Class))
  tax.table$Order <- ifelse(is.na(tax.table$Order) == TRUE , "Unknown", as.character(tax.table$Order))
  tax.table$Family <- ifelse(is.na(tax.table$Family) == TRUE , "Unknown", as.character(tax.table$Family))
  tax.table$Genus <- ifelse(is.na(tax.table$Genus) == TRUE , "Unknown", as.character(tax.table$Genus))
  tax.table$Species <- ifelse(is.na(tax.table$Species) == TRUE , "Unknown", as.character(tax.table$Species))
  tax.table$ASV <- rownames(tax.table)
  rownames(tax.table) <- NULL
  tax.table <- tax.table
}

```


```{r}
theme_set(theme_bw())
#set up color palette
farrowAndBall_palette <- c(
  "#4d5b6a" #Stiffkey Blue
  ,"#6a90b4" #Cook's Blue
  ,"#599ec4" #ST Giles Blue
  ,"#a1c5c8" #Blue Ground
  ,"#7997a1" #Stone Blue
  ,"#427e83" #Vardo
  ,"#84b59c" #Arsenic
  ,"#919f70" #Yeabridge Green
  ,"#686a47" #Bancha
  ,"#c8bd83" #Churlish Green
  ,"#cb9e59" #India Yellow
  ,"#ecc363" #Babouche
  ,"#c57b67" #Red Earth
  ,"#d65f3d" #Charlotte's Locks
  ,"#a04344" #Incarnadine
  ,"#bf7a8f" #Rangwali
  ,"#8d8089" #Brassica
  ,"#50414c" #Pelt
  ,"#e5e0db" #Strong White
  ,"#444546" #Off-Black
)
farrowAndBall_pal <- function() scales::manual_pal(farrowAndBall_palette)
scale_fill_farrowAndBall <- function(...) ggplot2::discrete_scale('fill', 'farrowAndBall', farrowAndBall_pal(), ...)
```

```{r}
#Removed ASVs without at least 1 read in at least 1 sample
mf_filter <- filter_taxa(sujan_biom, function(x) sum(x >= 1) >= 1, TRUE)

#Removed samples with no reads
mf_filter <- prune_samples(sample_sums(mf_filter) > 0, mf_filter)
mf_filter #removed 3 samples
#set order for SampleType
mf_filter@sam_data$Genotype <- factor(mf_filter@sam_data$Genotype,levels=c("ES4","UCR369"))


```

## Figure1-A
```{r}
richness = estimate_richness(mf_filter,measure="Shannon")
s <- data.frame(sample_data(mf_filter))
alphadiv <- cbind(richness, s)
alphadiv <- data.frame(alphadiv)
alphadiv$Drought_Stage <- factor(alphadiv$Drought_Stage,levels=c("V2","V4","R1","R4"))
alphadiv$Treatment <- factor(alphadiv$Treatment,levels=c("Control","Drought"))
alphadiv$Genotype <- factor(alphadiv$Genotype,levels=c("ES4","UCR369"))
```

```{r}
#statistical difference between categories
interac <- with(alphadiv, interaction(Genotype, Drought_Stage, Treatment))
anova_interac <- aov(data=alphadiv, formula = Shannon~interac)
summary(anova_interac)

out <- HSD.test(anova_interac, "interac", group=TRUE)

out
plot(out)
```

```{r}
#set the order of the three treatments along the x axis
alphadiv$Drought_Stage <- factor(alphadiv$Drought_Stage, levels=c("V2","V4","R1","R2"))

#taking the stats data from the out table and putting it into a dataframe
mf_filter@sam_data$Drought_Stage <- as.factor(mf_filter@sam_data$Drought_Stage)
mf_filter@sam_data$Genotype <- as.factor(mf_filter@sam_data$Genotype)
mf_filter@sam_data$Treatment <- as.factor(mf_filter@sam_data$Treatment)

vars <- data.frame(expand.grid(levels(mf_filter@sam_data$Drought_Stage),levels(mf_filter@sam_data$Genotype), levels(mf_filter@sam_data$Treatment)))

colnames(vars) <- c("Drought_Stage","Genotype", "Treatment")

write.csv(vars, file = "vars.csv")

dat <- read.csv("vars.csv")
dat$Shannon <- as.numeric(as.character(dat$Shannon))

dat$Drought_Stage <- factor(dat$Drought_Stage, levels = c("V2", "V4", "R1", "R4"))

alpha1 <- ggplot(alphadiv, aes(x = Drought_Stage, y = Shannon, fill = Treatment)) + 
  geom_boxplot(position = position_dodge(width = 0.75)) +
   geom_point(position = position_jitterdodge(), size = 2) +
  facet_grid(~Genotype, scales = "free", space = "free",
             labeller=labeller(Genotype=c("ES4"="ES4","UCR369"="UCR369")))+
  scale_fill_manual(breaks = c("Control", "Drought"),
                    values = c("#c57b67", "#84b59c")) +  # Corrected color values
  labs(title = "Shannon Diversity Index by Drought Stage and Genotype",
       x = "Drought Stage",
       y = "Shannon Diversity Index",
       fill = "Treatment") + 
  theme(axis.text.x = element_text(size = 14, color = "black", angle = 45, hjust = 1.0, vjust = 1.0), 
        axis.text.y = element_text(size = 16, color = "black"), 
        axis.title = element_text(size = 16, face = "bold"),
        text = element_text(size = 16)) +
  theme(strip.text = element_text(size = 14, face = "bold"))


alpha1

```
```{r}
ord1c <- plot_ordination(mf_filter, ordinate(mf_filter, "MDS"), axes=1:2, color = "Drought_Stage", shape = "Genotype") + 
  geom_point(size = 3)+
  scale_color_manual(values=c("red","blue","darkgreen", "purple"),
                     name = "Drought Stage",
                     breaks=c("V2","V4","R1","R4"),
                     labels=c("V2-Stage","V4-Stage","R1-Stage", "R4-Stage"))+
  scale_shape_manual(values = c(16, 17, 18, 19),  # Adjust shapes as needed
                     name = "Genotype") +
    theme(axis.text.x=element_text(size=14,color="black",angle=90), axis.text.y=element_text(size=14,color="black"), 
        axis.title=element_text(size=14),text=element_text(size=14))

ord1c
```

```{r}
#Indicator analyses are below along with scripts to make supporting files for tree-building
#Also makes dataframes that were used to make annotation files for iTOL
#iTOL was used to plot trees and change aesthetics
all <- import_biom("sujan_16s.biom")
metadata <- import_qiime_sample_data("metadata.txt")
tree <- read_tree("rooted-tree-sujan16s.nwk")
all <- merge_phyloseq(all, metadata, tree)
metadata <- read.delim("metadata.txt")
taxonomy <- read.delim("taxonomy.csv")
repseqs <- read.fasta(file = "fasta-sequences.fasta")

#this will merge all ASVs that have matching taxonomy assignments at the genus level
#it will retain, but not merge, ASVs whose assigned genera are either NA, "", " ", "\t", or "g__"
#takes a while...
all.genglom <- tax_glom(all, taxrank=rank_names(all)[6], 
                NArm = FALSE, 
                bad_empty = c(NA, "", " ", "\t", "g__"))

#remove ultralow abundance ASVs to prevent false postives in indicator analysis
#we're interested in ASVs that are reliable indicators
#an ASV with 10 (out of 10K!) reads in one or two samples in a category could be a significant indicator simply by virtue of not being found at all in other categories
#removes ASVs with fewer than 25 total reads and must be present in at least 1% of all samples (i.e. 4 samples in this case)
filter <- phyloseq::genefilter_sample(all.genglom, filterfun_sample(function(x) x >= 25), 
                                       A = 0.01*nsamples(all.genglom)-1)
all.genglom.filtered <- prune_taxa(filter, all.genglom)

#remove any samples with fewer than 2000 reads, then transform to relative abundance
all.genglom.filtered <- prune_samples(sample_sums(all.genglom.filtered)>=2000, all.genglom.filtered)
all.genglom.filtered <- transform_sample_counts(all.genglom.filtered, function(x) x/sum(x))
tax.table <- clean_taxtable(all.genglom.filtered)

all@sam_data$Drought_Stage <- as.factor(all@sam_data$Drought_Stage)
all@sam_data$Treatment <- as.factor(all@sam_data$Treatment)
all@sam_data$Genotype <- as.factor(all@sam_data$Genotype)

#run indicator analysis on each subset of data
SO.control <- subset_samples(all.genglom.filtered, Treatment == "Control")
SO.drought <- subset_samples(all.genglom.filtered, Treatment == "Drought")

SO.V2 <- subset_samples(all.genglom.filtered, Drought_Stage == "V2")
SO.V4 <- subset_samples(all.genglom.filtered, Drought_Stage == "V4")
SO.R1 <- subset_samples(all.genglom.filtered, Drought_Stage == "R1")
SO.R4 <- subset_samples(all.genglom.filtered, Drought_Stage == "R4")

sum(is.na(SO.control@sam_data$Treatment))
unique(SO.control@sam_data$Treatment)

SO.control@sam_data$Treatment <- as.factor(SO.control@sam_data$Treatment)
SO.drought@sam_data$Treatment <- as.factor(SO.drought@sam_data$Treatment)
SO.V2@sam_data$Drought_Stage <- as.factor(SO.V2@sam_data$Drought_Stage)

SO.Trt <- get_indicators(SO.control, groupfactor = "Treatment")
SO.Trt.drought <- get_indicators(SO.drought, groupfactor = "Treatment")

SO.V2i <- get_indicators(SO.V2, groupfactor = "Drought_Stage")


SO.Geno <- get_indicators(SO.bases, groupfactor = "Genotype")

#subset only  significant indicators
SO.tipIndicators$sig.indvals$subsection <- "tip"
SO.midIndicators$sig.indvals$subsection <- "mid"
SO.baseIndicators$sig.indvals$subsection <- "base"

#combine all subsections into the same df
SO.subIndicators <- rbind(SO.tipIndicators$sig.indvals, SO.midIndicators$sig.indvals, SO.baseIndicators$sig.indvals)
SO.subASVs <- unique(subset(SO.subIndicators, select = ASV))
#make_fasta(SO.subASVs, repseqs, "SO.subIndicators.fasta")

#make a csv to help build aestethic mappings for iTOL
SO.subIndicators.iTOL <- SO.subIndicators
SO.subIndicators.iTOL$group <- as.numeric(SO.subIndicators.iTOL$group)
SO.subIndicators.iTOL$group[SO.subIndicators.iTOL$group == 2] <- -1
SO.subIndicators.iTOL <- subset(SO.subIndicators.iTOL, select = c(ASV, group, subsection))
SO.subIndicators.iTOL <- recast(SO.subIndicators.iTOL, ASV~subsection, measure.var = "group")
SO.subIndicators.iTOL[is.na(SO.subIndicators.iTOL)] <- 0
SO.subIndicators.iTOL <- merge(SO.subIndicators.iTOL, tax.table, by = "ASV")
#write.csv(SO.subIndicators.iTOL, row.names = FALSE, quote = FALSE, file = "iTOL-files/SO-subIndicators-iTOL.csv")
```

## beta diversity patterns analysis

```{r}
# Calculate weighted UniFrac distances
unifrac_dist <- distance(mf_filter, method = "wunifrac")

# Perform PCoA
pcoa <- ordinate(mf_filter, method = "PCoA", distance = unifrac_dist)

# Extract the first axis values
axis1_values <- pcoa$vectors[, 1]

# Check the dimensions and structure of the extracted values
print(dim(pcoa$vectors))
print(head(pcoa$vectors))

# Convert to a data frame
# Convert to a data frame
axis1_df <- data.frame(SampleID = rownames(pcoa$vectors), Axis1 = axis1_values)

# Save to CSV
write.csv(axis1_df, "PCoA_Axis1_Values.csv", row.names = FALSE)

# Load the data from the CSV file
data <- read.csv("pattern_scores.csv")

# Ensure the DroughtStage column is treated as a factor (categorical variable)
data$Drought_Stage <- factor(data$Drought_Stage, levels = unique(data$Drought_Stage))

data$Genotype <- factor(data$Genotype, levels = unique(data$Genotype))

# Create a ggplot with Axis1 on the y-axis and DroughtStage on the x-axis
ggplot(data, aes(x = Drought_Stage, y = Axis1, color = Treatment, shape = Genotype)) +
  geom_point(size = 3) + # Use points to plot the data
  theme_minimal() + # Use a minimal theme
  labs(title = "PCoA Axis 1 vs. Drought Stages",
       x = "Drought Stages",
       y = "PCoA Axis 1") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) # Rotate x-axis labels if needed
```


#Workshop-Advanced microbiome

```{r}
plot_richness(physeq = mf_filter, x="Drought_Stage", color = "Genotype", measures = c("Observed", "Chao1", "Shannon")) + geom_boxplot() +theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

```


## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
